cfme/tests/containers/test_relationships_tables.py

```diff
@@ -8,252 +7,73 @@
-        table_locator="//div[@id='list_grid']//table")
-    ui_images = [r.name.text for r in list_tbl_image.rows()]
-
-    for name in ui_images:
-        img = Image(name, provider)
-        navigate_to(img, 'Details')
-
-        val = img.get_detail('Relationships', rel)
-        assert val != 'Unknown image source'
-
-        sel.click(InfoBlock.element('Relationships', rel))
-
-        # Containers Provider and Image Registry are string values
-        # Other rows in the table show the number of the given items
-        if rel in ['Containers Provider', 'Image Registry']:
-            assert val == InfoBlock.text('Properties', detailfield)
-        else:
-            val = int(val)
-            # There might be more than 1 page of items
-            if paginator.page_controls_exist():
-                paginator.results_per_page(1000)
-            assert len([r for r in list_tbl_image.rows()]) == val
-
-
-# 9868 #9869
-
-
-@pytest.mark.parametrize('rel',
-                         ['Containers Provider',
-                          'Routes',
-                          'Services',
-                          'Replicators',
-                          'Pods',
-                          'Nodes'])
-def test_projects_rel(provider, rel):
-    sel.force_navigate('containers_projects')
+    navigate_to(cls, 'All')
     tb.select('List View')
-    list_tbl_project = CheckboxTable(
-        table_locator="//div[@id='list_grid']//table")
-    ui_projects = [r.name.text for r in list_tbl_project.rows()]
-    mgmt_objs = provider.mgmt.list_project()
-
-    assert set(ui_projects).issubset(
-        [obj.name for obj in mgmt_objs]), 'Missing objects'
-
-    for name in ui_projects:
-        obj = Project(name, provider)
-
-        val = obj.get_detail('Relationships', rel)
-        if val == '0':
-            continue
-        obj.click_element('Relationships', rel)
-
-        try:
-            val = int(val)
-            assert len([r for r in list_tbl_project.rows()]) == val
-        except ValueError:
-            assert val == InfoBlock.text('Properties', 'Name')
+    list_tbl = CheckboxTable(table_locator="//div[@id='list_grid']//table")
+    cls_instances = [r.name.text for r in list_tbl.rows()]
+    cls_instances_revised = filter(
+        lambda ch: 'nginx' not in ch and not ch.startswith('metrics'),
+        cls_instances)
+    for name in cls_instances_revised:
+        navigate_to(cls, 'All')
+        obj = cls(name, provider)
+        rel_tbl = obj.summary.groups()['relationships']
+        keys = [key for key in rel_tbl.keys]
+        for key in keys:
+            # reload summary to prevent StaleElementReferenceException:
+            obj.summary.reload()
+            rel_tbl = obj.summary.groups()['relationships']
+            element = rel_tbl.__getattribute__(key)
+            value = element.value
+            if value in rel_values:
+                continue
+            sel.click(element)
+
+            try:
```

Comment:
I am assuming only the line 68 should be protected so this should rather be:

            try:
                value = int(value)
            except ValueError:
                assert value == details_page.infoblock.text(
                    'Properties', 'Name')
            else:
                # best effort to include all items from rel in one page
                if paginator.page_controls_exist():
                    paginator.results_per_page(1000)
                else:
                    logger.warning(
                        'Unable to increase results per page, '
                        'assertion against number of rows may fail.')
                assert len([r for r in list_tbl.rows()]) == value

The http://stackoverflow.com/a/31626974/1950100 is the best explanation for the try-except-else-finally which I have found so far.